Frends.AmazonS3.UploadObject/Frends.AmazonS3.UploadObject.Test/UnitTest1.cs

```diff
@@ -0,0 +1,916 @@
+        /// </summary>
+        [TestMethod]
+        public void AWSCredsACLBucketOwnerFullControl()
+        {
+            _input = new Input
+            {
+                FilePath = $@"{_dir}\AWS",
+                ACL = ACLs.BucketOwnerFullControl,
+                FileMask = null,
+                UseACL = true,
+                S3Directory = "2022/AWSCreds/ACLBucketOwnerFullControl/",
+            };
+            _connection = new Connection
+            {
+                AuthenticationMethod = AuthenticationMethod.AWSCredentials,
+                PreSignedURL = null,
+                AwsAccessKeyId = _accessKey,
+                AwsSecretAccessKey = _secretAccessKey,
+                BucketName = _bucketName,
+                Region = Region.EuCentral1,
+            };
+            _options = new Options
+            {
+                UploadFromCurrentDirectoryOnly = false,
+                Overwrite = false,
+                PreserveFolderStructure = false,
+                ReturnListOfObjectKeys = false,
+                DeleteSource = false,
+                ThrowErrorIfNoMatch = false,
+            };
+
+            var result = AmazonS3.UploadObject(_connection, _input, _options, default);
+            Assert.IsNotNull(result.Result.Results);
+        }
+        #endregion AWSCredsACLBucketOwnerFullControl
+
+        #region AWSCredsACLLogDeliveryWrite
+        /// <summary>
+        /// AWS creds testing.
+        /// ACL LogDeliveryWrite.
+        /// </summary>
+        [TestMethod]
+        public void AWSCredsACLLogDeliveryWrite()
+        {
+            _input = new Input
+            {
+                FilePath = $@"{_dir}\AWS",
+                ACL = ACLs.LogDeliveryWrite,
+                FileMask = null,
+                UseACL = true,
+                S3Directory = "2022/AWSCreds/ACLLogDeliveryWrite/",
+            };
+            _connection = new Connection
+            {
+                AuthenticationMethod = AuthenticationMethod.AWSCredentials,
+                PreSignedURL = null,
+                AwsAccessKeyId = _accessKey,
+                AwsSecretAccessKey = _secretAccessKey,
+                BucketName = _bucketName,
+                Region = Region.EuCentral1,
+            };
+            _options = new Options
+            {
+                UploadFromCurrentDirectoryOnly = false,
+                Overwrite = false,
+                PreserveFolderStructure = false,
+                ReturnListOfObjectKeys = false,
+                DeleteSource = false,
+                ThrowErrorIfNoMatch = false,
+            };
+
+            var result = AmazonS3.UploadObject(_connection, _input, _options, default);
+            Assert.IsNotNull(result.Result.Results);
+        }
+        #endregion AWSCredsACLLogDeliveryWrite
+
+        #endregion AWS Creds
+
+        #region HelperMethods
+
+        /// <summary>
+        /// Create test files and folders.
+        /// </summary>
+        [TestInitialize]
```

Comment:
These are not helper methods, but setup and cleanup methods. Setups and cleanups should be before test methods, so move these before tests.